modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/model/listener/APIEndpointRelevantObjectEntryModelListener.java

```diff
@@ -237,14 +237,19 @@ private void _validateAPISchema(long apiApplicationId, long apiSchemaId)
 	}
 
 	private void _validateSingleElementPath(
-			ObjectEntry objectEntry, String pathParameter, String pathString)
+			ObjectEntry objectEntry, String pathParameter, String pathString,
+			long responseAPISchemaId)
 		throws Exception {
 
-		if (Validator.isNull(pathParameter)) {
+		if ((pathParameter != null) && (responseAPISchemaId != 0) &&
+			!Objects.equals(pathParameter, "") &&
```

Comment:
I think we could use a more specific method for Strings to avoid include two conditions != null and != ""